rts/lra/client/src/main/java/io/narayana/lra/client/NarayanaLRAClient.java

```diff
@@ -734,18 +738,20 @@ private URI enlistCompensator(URI uri, Long timelimit, String linkHeader, String
 
             String responseEntity = response.hasEntity() ? response.readEntity(String.class) : "";
             if (response.getStatus() == Response.Status.PRECONDITION_FAILED.getStatusCode()) {
-                LRALogger.i18nLogger.error_tooLateToJoin(lraId, responseEntity);
-                String errorMsg = lraId + ": Too late to join with this LRA";
-                throw new WebApplicationException(errorMsg,
-                        Response.status(PRECONDITION_FAILED).entity(errorMsg).build());
+                String logMsg = LRALogger.i18nLogger.error_tooLateToJoin(lraId, responseEntity);
+                LRALogger.logger.error(logMsg);
+                throw new WebApplicationException(logMsg,
+                        Response.status(PRECONDITION_FAILED).entity(logMsg).build());
             } else if (response.getStatus() == NOT_FOUND.getStatusCode()) {
-                LRALogger.i18nLogger.info_failedToEnlistingLRANotFound(
+                String logMsg = LRALogger.i18nLogger.info_failedToEnlistingLRANotFound(
                         lraId, coordinatorUrl, NOT_FOUND.getStatusCode(), NOT_FOUND.getReasonPhrase(), GONE.getStatusCode(), GONE.getReasonPhrase());
-                throw new WebApplicationException(uri.toASCIIString(),
+                LRALogger.logger.info(logMsg);
+                throw new WebApplicationException(logMsg,
```

Comment:
Part of what we discussed on this LRA is that the message in the exception response needs to correspond to the i18n log message. Here you are setting the exception message to be the same as the log message but you are then passing something different back to the caller, if you check the [javadoc for this exception](https://docs.oracle.com/javaee/7/api/javax/ws/rs/WebApplicationException.html#WebApplicationException-java.lang.String-javax.ws.rs.core.Response-) you should see what I mean.

I also said that the value in the uri variable is the same as the value in the lraId so the value in the response entity adds nothing, so this exception does not follow this advice. The intention is for the log message and the exception response to be consistent - you have followed this advice in other parts of this PR but we need consistency across the whole of the PR.

This looks like the only remaining issue on the PR so we are almost there!